### PR TITLE
Implementation of integration test coverage

### PR DESCRIPTION
**Pull request by ruflin**

### CODE REVIEW

**Comment by andrewkroh** [comment]
Nice work! Now the coverage number looks even better.

File: Makefile [hunk 1]
```diff
@@ -117,5 +117,20 @@ gen:
...
+# Generates packetbeat.test coverage testing binary
+test-binary:
+	$(GODEP) go test -c -cover -covermode=count -coverpkg ./...
+
+full-coverage:
+	make coverage
+	make -C ./tests coverage
+	# Writes count mode on top of file
+	echo 'mode: count' > ./coverage/full.cov
+	# Collects all integration coverage files and skipts top line with mode
```
Comment (andrewkroh) [inline Makefile]: `s/skipts/skips/`

Comment (ruflin) [inline Makefile]: Fixed

File: tests/Makefile [hunk 2]
```diff
@@ -10,5 +10,21 @@ env/bin/activate: requirements.txt
...
 	. env/bin/activate; nosetests --processes=${PROCESSES} --process-timeout=$(TIMEOUT)
+
+.PHONY: coverage
+coverage: test
+	mkdir -p ../coverage
+	# Writes count mode on top of file
+	echo 'mode: count' > ../coverage/integration.cov
+	# Collects all integration coverage files and skipts top line with mode
+	tail -q -n +2 ./run/**/*.cov >> ../coverage/integration.cov
+	# Generats HTML output
```
Comment (tsg) [inline tests/Makefile]: typo in comment here: s/Generats/Generates/

File: tests/Makefile [hunk 3]
```diff
@@ -10,5 +10,21 @@ env/bin/activate: requirements.txt
...
-	make -C ..
+	make -C .. test-binary
 	. env/bin/activate; nosetests --processes=${PROCESSES} --process-timeout=$(TIMEOUT)
+
+.PHONY: coverage
+coverage: test
+	mkdir -p ../coverage
+	# Writes count mode on top of file
+	echo 'mode: count' > ../coverage/integration.cov
+	# Collects all integration coverage files and skipts top line with mode
```
Comment (tsg) [inline tests/Makefile]: typo in comment: s/skipts/skips/

File: Makefile [hunk 4]
```diff
@@ -117,5 +117,20 @@ gen:
...
-	rm packetbeat || true
-	rm -r packetbeat-$(VERSION) || true
+	-rm packetbeat
+	-rm packetbeat.test
+	-rm -r packetbeat-$(VERSION)
+	-rm -r coverage
+
+# Generates packetbeat.test coverage testing binary
+test-binary:
+	$(GODEP) go test -c -cover -covermode=count -coverpkg ./...
```
Comment (tsg) [inline Makefile]: I suggest to name the target packetbeat.test and make it depend on the $(GOFILES) just like the Packetbeat one. Alternatively, declare it as .PHONY, but I think first version is  better.

**Comment by tsg** [comment]
Nice, LGTM apart from the minor comments above. This is likely the coolest Golang related hack I've seen in the last few months :-).

File: Makefile (hunk 4 above)
Comment (ruflin) [inline Makefile]: Does the dependency on $(GOFILES) mean it is only rebuilt in case on of the files changed? Not sure why this replaces the .PHONY part?

**Comment by ruflin** [comment]
@tsg Typos fixed and target renamed.

**Pull request merged**